internal/antewrapper/msg_fees_decorator.go

```diff
@@ -238,59 +238,68 @@ func checkFloorGasFees(gas uint64, feeCoins sdk.Coins, additionalFees sdk.Coins,
 	if _, hasNeg := feeCoins.SafeSub(sdk.Coins{baseFees}); hasNeg {
 		// for tx without additional fees.
 		if additionalFees == nil || additionalFees.IsZero() {
-			return sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "not enough fees based on floor gas price: %q; required base fees >=%q: Supplied fee was %q", minGasPriceForAdditionalFeeCalc, baseFees, feeCoins)
+			return sdkerrors.ErrInsufficientFee.Wrapf("not enough fees based on floor gas price: %q; required base fees >=%q: Supplied fee was %q", minGasPriceForAdditionalFeeCalc, baseFees, feeCoins)
 		}
-		return sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "not enough fees based on floor gas price: %q; after deducting (total fee supplied fees - additional fees(%q)) required base fees >=%q: Supplied fee was %q", minGasPriceForAdditionalFeeCalc, additionalFees, baseFees, feeCoins)
+		return sdkerrors.ErrInsufficientFee.Wrapf("not enough fees based on floor gas price: %q; after deducting (total fee supplied fees - additional fees(%q)) required base fees >=%q: Supplied fee was %q", minGasPriceForAdditionalFeeCalc, additionalFees, baseFees, feeCoins)
 	}
 	return nil
 }
 
-// CalculateAdditionalFeesToBePaid computes the stability tax on MsgSend and MsgMultiSend.
+// CalculateAdditionalFeesToBePaid computes the additional fees to be paid and the distributions to recipients
 func CalculateAdditionalFeesToBePaid(ctx sdk.Context, mbfk msgfeestypes.MsgFeesKeeper, msgs ...sdk.Msg) (*MsgFeesDistribution, error) {
-	// get the msg fee
 	msgFeesDistribution := MsgFeesDistribution{
 		RecipientDistributions: make(map[string]sdk.Coin),
 	}
-	assessCustomMsgTypeURL := sdk.MsgTypeURL(&msgfeestypes.MsgAssessCustomMsgFeeRequest{})
 	for _, msg := range msgs {
 		typeURL := sdk.MsgTypeURL(msg)
 		msgFees, err := mbfk.GetMsgFee(ctx, typeURL)
 		if err != nil {
-			return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+			return nil, sdkerrors.ErrInvalidRequest.Wrap(err.Error())
 		}
 
 		if msgFees != nil {
 			if msgFees.AdditionalFee.IsPositive() {
-				msgFeesDistribution.AdditionalModuleFees = msgFeesDistribution.AdditionalModuleFees.Add(msgFees.AdditionalFee)
-				msgFeesDistribution.TotalAdditionalFees = msgFeesDistribution.TotalAdditionalFees.Add(msgFees.AdditionalFee)
+				if err := CalculateDistributions(msgFees.Recipient, msgFees.AdditionalFee, msgFees.RecipientBasisPoints, &msgFeesDistribution); err != nil {
+					return nil, err
+				}
 			}
 		}
-		if typeURL == assessCustomMsgTypeURL {
+		if typeURL == sdk.MsgTypeURL(&msgfeestypes.MsgAssessCustomMsgFeeRequest{}) {
```

Comment:
The `assessCustomMsgTypeURL` variable was used so that we aren't repeatedly calling `sdk.MsgTypeURL` in a loop.

Reply:
Duh, I was the one who put that in there in the first place.  Second time around thought it was inside the loop so it was redundant.  😶‍🌫️